### DISCUSSION TITLE
Suggestion: Flat docks

### DISCUSSION THREAD

**Discussion by Limyx826** [OpenTTD/OpenTTD#12916] [Features]
Someone did try for a NewGRF on flat docks some years ago, but that project is going nowhere. I ain't sure is it possible to code that with NewGRF or require new feature coding on the base game source code directly.

**Comment by LC-Zorg** [comment]
> I ain't sure is it possible to code that with NewGRF or require new feature coding on the base game source code directly.

Reading the tea leaves, I can say that it would almost certainly have to be a new feature. ;)

For me, it's one of the most anticipated features. And as I noticed, not only for me. :) [Pony wishes from andythenorth](https://www.tt-forums.net/viewtopic.php?p=1181978#p1181978) (edition [2017](https://www.tt-forums.net/viewtopic.php?p=1181978#p1181978), edition [2021](https://www.tt-forums.net/viewtopic.php?p=1240361#p1240361), ...). 
From what Michael Blunck writes, this feature supposedly existed in TTD Patch.

A few of my visualizations of what it could look like. ;)

![Flat docks parallel to the quay - both types](https://github.com/user-attachments/assets/e2ebe7f1-7d86-4f8f-bd90-c644d86d2668)

![Flat docks parallel to the quay - ferry dock](https://github.com/user-attachments/assets/12c1e3d7-f0b9-4dc0-8b97-feeed2427f4a)

![Flat docks parallel to the quay - mooring path](https://github.com/user-attachments/assets/c0f78394-8cec-40c3-a21c-af6eb52612ab)

![Flat docks parallel to the quay - differentiation of cargo and ship sizes](https://github.com/user-attachments/assets/1a7ebdc4-8eb0-4f04-bbed-0f8c02cad62d)

![Flat docks parallel to the quay - interface](https://github.com/user-attachments/assets/27f374c3-6760-45d6-9bee-226d3b49a17e)

![Flat docks parallel to the quay - new settings](https://github.com/user-attachments/assets/f85098a3-2e0c-4c69-8952-ab0d66d953c2)
